File: Sources/FWObjC/Service/Basic/FWLanguage.m
//
//  FWLanguage.m
//  FWFramework
//
//  Created by wuyong on 2022/8/22.
//

#import "FWLanguage.h"

NSNotificationName const FWLanguageChangedNotification = @"FWLanguageChangedNotification";

#if FWMacroSPM



#else

#import <FWFramework/FWFramework-Swift.h>

#endif

#pragma mark - NSString+FWLanguage

@implementation NSString (FWLanguage)

- (NSString *)fw_localized
{
    return [NSBundle fw_localizedString:self table:nil];
}

@end

File: FWFramework/Classes/Swift/Service/FWLanguage.swift
//
//  FWLanguage.swift
//  FWFramework
//
//  Created by wuyong on 2019/6/27.
//  Copyright © 2019 wuyong.site. All rights reserved.
//

import Foundation

/// 读取本地化字符串
///
/// - Parameters:
///   - key: 本地化键名
///   - table: 本地化表名，默认Localizable.strings
/// - Returns: 本地化字符串
public func FWLocalizedString(_ key: String, _ table: String? = nil) -> String {
    return Bundle.fwLocalizedString(key, table: table)
}
